### PR TITLE
Add more backoff from start/update errors

### PR DESCRIPTION
**Pull request by smondet**
This is for #95.

The current “formula” for the backoff is `number-of-errors × 30 seconds`,
that maybe too much.

### CODE REVIEW

File: src/lib/server.ml [hunk 1]
```diff
@@ -196,29 +196,53 @@ let rec loop:
...
             | `Started _ -> c + 1
             | _ -> c) in
       let max_started = Cluster.max_started_jobs t.cluster in
-      List.fold t.jobs ~init:(`Started currently_started, [])
-        ~f:(fun (`Started started, todo) j ->
+      let reupdate_wait j =
+        match Job.latest_error j with
+        | None -> 30.
+        | Some t ->
+          (* let time_ago = now () -. t in *)
```
Comment (ihodes) [inline src/lib/server.ml]: old line can be rm'd

**Comment by ihodes** [comment]
LGTM, thanks for adding this!

**Comment by smondet** [comment]
So @ihodes what should be the default "backoff" formula? what should be a parameter?

It turns out with the WIP one here it gets to the job failing half-an-hour after submission (and I tried with a legitimate error: unknown docker image).

**Comment by ihodes** [comment]
Yeah that's kinda tricky; maybe the params are just number of tries, which is already a param (which is 10 right now), and we can set a new default to 5 or something?

**Comment by smondet** [comment]
So with a given `factor`, and `retries` retries, a job that fails to
start will actually fail after:

```
1 × factor + 2 × factor + ... + retries × factor + (retries + 1) × factor
```

(the `retries+1`th being the fatal one).

So that's

```
factor × (retries + 1) × (retries + 1 + 1) / 2
```

With default values (factor = 30, retries = 10): 33 minutes.

**Comment by smondet** [comment]
OK @ihodes removed the `WIP` label :)

**Comment by ihodes** [comment]
This SGTM; I'd still change the default retry # to maybe 7, so we have about 10 minutes of waiting on kubernetes instead of like 17 with the 20s backoff factor?

**Pull request merged**